sklearn/utils/tests/test_stats.py

```diff
@@ -74,14 +104,22 @@ def test_weighted_median_integer_weights():
     assert median == approx(w_median)
 
 
-def test_weighted_percentile_2d():
+@pytest.mark.parametrize(
+    "array_namespace, device, dtype_name", yield_namespace_device_dtype_combinations()
+)
+def test_weighted_percentile_2d(array_namespace, device, dtype_name):
+    xp = _array_api_for_tests(array_namespace, device)
     # Check for when array 2D and sample_weight 1D
     rng = np.random.RandomState(0)
     x1 = rng.randint(10, size=10)
```

Comment:
In case we parametrise this test with Array API, `x1` would need to be xp-isized like the other data.

Reply:
I don't think so. `x1` is only an intermediate step to generate `x_2d` via `np.vstack(x1, x2)`. The result of the stacking (`x_2d`) is converted to the `xp` namespace before calling `_weighted_percentile`.

Only the arguments of `_weighted_percentile` need to be moved to the `xp` namespace. The way those arguments are constructed using numpy does not matter.